### DISCUSSION TITLE
Notify-push container is "Waiting for Nextcloud to start ..."

### DISCUSSION THREAD

**Discussion by Gavin152** [nextcloud/all-in-one#3156] [Questions] [answered]
On a fairly fresh AIO install (via docker) I cannot get push notifications to work. The logs of the `nextcloud-aio-notify-push` only state `Waiting for Nextcloud to start...` over and over again. NC is running behind a Caddy reverse proxy that has been setup according to the reverse-proxy guide (and seems to work fine). 

Here's the NC part of my caddy config: 
```
my.public.domain {
	handle {
		reverse_proxy localhost:11000
	}
}
```

### Steps to reproduce
1. Installed AIO instance
2. Installed NC app on android and logged in with admin user (all seems to work fine here, eg access to files etc)
3. Enable push notification (and email) for changes in calendar events and when files or folders are changed. (green toastr message shows that new settings have been persisted successfully)
4. change something on a calendar event or a file (tried create/edit/delete)
5. hope for push notification on phone

### Expected behavior 
6. receive push notification on phone

### Actual behavior 
6. DON'T receive push notification on phone

### Host OS 
AIO docker install on Debian 5.10.x
Docker version: 24.0.5

#### Nextcloud AIO version 
Cannot find this info on the AIO Interface, but according to the enabled apps section in the system report its:
 - nextcloud-aio: 0.4.0
 - notifications: 2.14.0
 - notify_push: 0.6.3

#### Other valuable info 
While digging around and trying to familiarize myself with the NC container landscape I found the `nextcloud-aio-notify-push` container, which seems to be a rather recent development [PR #2703](https://github.com/nextcloud/all-in-one/pull/2703). 
I saw that somehow the application inside the container is waiting for a connection to NC to be established. All that I'm seeing in the logs is `Waiting for Nextcloud to start...` over and over again. 
I also tried restarting the container, but no luck.

I can submit a System Report if needed.

Is there a setup step that I'm missing to get this to work?

**Comment by szaimen** [comment]
> 6\. DON'T receive push notification on phone

Push notifications are not sent via notify-push but instead via push.nextcloud.com. See https://help.nextcloud.com/t/notify-push-invalid-credentials-in-nextcloud-aio/168052/4 for the solution.

**Reply by Gavin152** [reply]
I'm sorry to report, that this didn't fix it for me. 

However, with the linked thread I found out, that I hadn't setup the `trusted_proxies` directive. This resulted in the notify-push self test to fail. Now with the `trusted_proxies` setup, the self test succeeds. But there's still no notification, neither pushed nor via email (emailing is setup and the test mail works).

Unfortunately I cannot find anything in the container logs or under Administration =>Logging that points towards an attempt of a notification being sent.

**Comment by szaimen** [comment]
> neither pushed

As I saidx this is sent via push.nextcloud.com.

Did you already recreate your account in the android/ios app?

**Reply by Gavin152** [reply]
Yes, twice. the last time after having the self-test succeed. I removed the account via Manage Accounts => Remove Account and the re-logged in. 

The Activities tab shows `You created an app password for a session named "My phone signature (Android)"` at the appropriate moment.

**Reply by szaimen** [reply]
Can you activate the daily backup in the aio interface and let it run now (by setting the time) and check if you have received a push notification afterwards about a successfull backup?

**Reply by Gavin152** [reply]
Just tried, but clicking the link to the AIO IF results in the request to `getlogin?token=<my_token>` to return a `SLIM application error` with the warning saying
```
Undefined array key "AIO_TOKEN" in <b>/var/www/docker-aio/php/src/Data/ConfigurationManager.php</b>
on line
```

I also tried the notification setup with a different (newly created) user, but no dice.

**Reply by szaimen** [reply]
Hm... it looks like you have broken the automatic login somehow. Please use the normal login via https://ip.address.of.the.server:8080/login instead. (you probably need to stop the apache container first in order to log in)

**Reply by Gavin152** [reply]
OK, I tried. I shutdown the apache container and headed to the login page. All requests come back with a 200, but all that happens is that the page reloads.

I'm starting to suspect, that there's a bigger underlying issue in my installation. When I head to the root of `my_server_ip:8080` I see a warning at the top of the page 
```
Warning: file_put_contents(/mnt/docker-aio-config/data//configuration.json): Failed to open stream: Permission denied in /var/www/docker-aio/php/src/Data/ConfigurationManager.php on line 507
```
Looks like an error in the path with the double `//` between `data` and `configuration`. I checked in the container and the `data` dir is empty.
```
/var/www/docker-aio # ls -alh /mnt/docker-aio-config/data/
total 8K     
drwxr-xr-x    2 root     root        4.0K Aug 12 09:55 .
drwxrwxrwx    7 root     root        4.0K Aug 12 09:56 ..
/var/www/docker-aio # 
```

__EDIT__
I checked the `/mnt/docker-aio-config` dir. Not all dirs belong to `www-data`. Don't know how this is intended to look.
```
/var/www/docker-aio # ls -alh /mnt/docker-aio-config/
total 24K    
drwxrwxrwx    6 root     root        4.0K Aug 14 12:54 .
drwxr-xr-x    1 root     root        4.0K Aug 12 00:04 ..
drwxrwx---    2 www-data www-data    4.0K Aug 12 00:02 caddy
drwxrwx---    2 root     root        4.0K Aug 12 00:02 certs
drwxr-xr-x    2 root     root        4.0K Aug 12 09:55 data
drwxrwx---    2 www-data www-data    4.0K Aug 14 12:44 session
/var/www/docker-aio # 
```

**Reply by szaimen** [reply]
Did you somehow change the permission of the underlying docker volume or broke your docker installation somehow? You could try to fix the permissions by by running `sudo chown -R www-data:www-data` on the volume...

**Reply by Gavin152** [reply]
OK, was able to fix the permissions on the volume, as well as failing authentication to the DB. I was able to create an initial backup and then a daily at a set time. That worked fine, but still there was no notification.

**Reply by szaimen** [reply]
Cool! Is the notification shown in the web interface?

**Reply by Gavin152** [reply]
Yes, notifications appeared in the web interface at the bell icon. But I still didn't get the notification on my phone.

**Reply by szaimen** [reply]
What kind of phone is this? Which OS does it run?

**Reply by Gavin152** [reply]
Running Android v13 on a Samsung S22 (that has One UI v5.1 on it) and the NextCloud app is v3.25.0. Notifications have been allowed and are enabled for the app, of course ;-)

**Reply by szaimen** [reply]
Hm... There should push notifications usually work. Did you install the nextcloud app from the play-store?

**Reply by Gavin152** [reply]
Yes, even tried re-installing by now. But still no luck.

Are there logs available somewhere, where one could see what's going on while a "notifyable" event is being emitted? Kinda feels like poking around in the dark and I don't wanna take up so much of your time.

**Reply by szaimen** [reply]
You could trigger a push notification with `sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ notification:generate "$userid" "summary" -l "content"` which simplifies testing.

**Reply by Gavin152** [reply]
OK, this was very helpful. Was able to get some notifications as expected on my phone. However, the notifications only popped up on my phone, when I opened the app. Not sure if this is intended this way. And even then, not all notifications triggered an os-level notification. Some only showed up in the app's notification tab, others popped up in both.

**Reply by szaimen** [reply, marked as answer]
> OK, this was very helpful. Was able to get some notifications as expected on my phone. However, the notifications only popped up on my phone, when I opened the app. Not sure if this is intended this way. And even then, not all notifications triggered an os-level notification. Some only showed up in the app's notification tab, others popped up in both.

Sounds like an issue within the android app then. Please refer here: https://github.com/nextcloud/android/issues

**Reply by Gavin152** [reply]
Thanks so much for the assistance, mate!

**Comment by JoseRamirezMoreno** [comment]
Solved it as follows:

sudo docker exec -it nextcloud-aio-nextcloud /bin/bash
cd /var/www/html/custom_apps/notify_push/bin/x86_64
chmod +x notify_push
